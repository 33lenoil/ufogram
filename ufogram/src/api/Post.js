import axios from 'axios';

export default async function post(user, image, caption) {
    try {
        const response = await axios.post('http://localhost:3000/posts', {
        image: image,
        caption: caption,
    });
    if (response.status === 201) {
        return response.data;
    } else {
        return 'Error: Post creation failed';
    }
    } catch(e) {
        console.error('Error:', e.message);
        throw e;
    }
}