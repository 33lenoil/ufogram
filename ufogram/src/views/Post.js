import React, { useState } from 'react';

function PostRender(props) {
    let [likes, setLikes] = useState(0);
    let [liked, setLiked] = useState(false);

    // const likeButton = document.getElementById("likeButton");

    const handleLike = ((clickEvent) => {
        if (liked) {
            setLikes(likes - 1);
            setLiked(false);
            clickEvent.target.innerHTML = "Like";
        } else {
            setLikes(likes + 1);
            setLiked(true);
            clickEvent.target.innerHTML = "Unlike";

        }
    })
    return (
        <div style={{margin: "10px"}}>
            <div>{ props.username }</div>
            {
                props.imageUrl.includes("youtube") && <iframe width="560" height="315" src={ props.imageUrl }></iframe>
            }
            {
                !props.imageUrl.includes("youtube") && <img src={ props.imageUrl } alt='image_unloaded' />
            }
            <div>{ props.caption }</div>
            <div>
                <button id="likeButton" onClick = { handleLike }>Like</button>  
            </div>
            <div>Likes: { likes }</div>
        </div>
    )
}

export default PostRender;