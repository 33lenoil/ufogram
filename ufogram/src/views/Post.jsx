import React from 'react';
import PropTypes from 'prop-types';

function PostRender({ username, imageUrl, caption }) {
//   const [likes, setLikes] = useState(0);
//   const [liked, setLiked] = useState(false);

  // const likeButton = document.getElementById("likeButton");

  //   const handleLike = ((clickEvent) => {
  //     console.log('1', liked);
  //     if (liked) {
  //       setLikes(likes - 1);
  //       setLiked(false);
  //       console.log('5', liked);
  //     //   clickEvent.target.innerHTML = 'Like';
  //     } else {
  //       setLikes(likes + 1);
  //       setLiked(true);
  //       console.log('6', liked);
  //     //   clickEvent.target.innerHTML = 'Unlike';
  //     }
  //     const button = clickEvent.target;
  //     console.log('2', button.innerHTML);
  //     button.innerHTML = liked ? 'Unlike' : 'Like';
  //     console.log('4', button.innerHTML);
  //   });
  return (
    <div style={{ margin: '10px' }}>
      <div>{ username }</div>
      {
                imageUrl.includes('youtube') && <iframe title="YouTube Video" width="560" height="315" src={imageUrl} />
            }
      {
                !imageUrl.includes('youtube') && <img src={imageUrl} alt="image_unloaded" />
            }
      <div>{ caption }</div>
      {/* <div>
        <button type="button" id="likeButton" onClick={handleLike}>Like</button>
      </div> */}
      {/* <div>
        Likes:
        {' '}
        { likes }
      </div> */}
    </div>
  );
}

// Prop validation using ESLint's prop object
PostRender.propTypes = {
  username: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  caption: PropTypes.string.isRequired,
};

export default PostRender;
