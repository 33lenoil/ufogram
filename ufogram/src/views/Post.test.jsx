/**
 * @jest-environment jsdom
 */

import React, { useState } from 'react';
import '@testing-library/jest-dom';
import {
  render, screen, act, waitFor,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Post from './Post';

test('renders like button', () => {
  render(
    <Router>
      {/* <Routes>
                <Route path="/signup" element={<Signup />} />
            </Routes> */}
      <Post username="lionelhu" imageUrl="https://picsum.photos/200/302" caption="Haha" />
    </Router>,
  );
  const linkElement = screen.getByRole('button', {
    name: /Like/,
  });
  expect(linkElement).toBeInTheDocument();
});

test('renders likes label', () => {
  render(
    <Router>
      {/* <Routes>
              <Route path="/signup" element={<Signup />} />
          </Routes> */}
      <Post username="lionelhu" imageUrl="https://picsum.photos/200/302" caption="Haha" />
    </Router>,
  );
  const linkElement = screen.getByText(/Likes:/);
  expect(linkElement).toBeInTheDocument();
});

/**
 * Snapshot Testing
 */

test('the component matches the snapshot', () => {
  const component = renderer.create(<Router>
    <Post username="lionelhu" imageUrl="https://picsum.photos/200/302" caption="Haha" />
  </Router>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 *  Testing
 */

// test('like button testing', async () => {
//   act(() => {
//     render(
//       <Router>
//         <Post username="lionelhu" imageUrl="https://picsum.photos/200/302" caption="Haha" />
//       </Router>,
//     );
//   });

//   // Find the "Like" button by its text
//   const likeButton = screen.getByRole('button', {
//     name: /Like/,
//   });

//   // Check if the "Like" button is in the document
//   expect(likeButton).toBeInTheDocument();

//   // Click the "Like" button
//   act(() => {
//     userEvent.click(likeButton);
//   });

//   // After clicking, check if the "Unlike" button is in the document
//   const unlikeButton = await screen.findByText(/Unlike/);

//   expect(unlikeButton).toBeInTheDocument();

//   // Click the "Unlike" button
//   act(() => {
//     userEvent.click(unlikeButton);
//   });

//   // After clicking, check if the "Like" button is back in the document
//   const likeButtonAgain = await screen.findByRole('button', {
//     name: /Like/,
//   });

//   expect(likeButtonAgain).toBeInTheDocument();
// });
