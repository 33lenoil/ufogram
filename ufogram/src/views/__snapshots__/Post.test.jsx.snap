// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the component matches the snapshot 1`] = `
<div
  style={
    Object {
      "margin": "10px",
    }
  }
>
  <div>
    lionelhu
  </div>
  <img
    alt="image_unloaded"
    src="https://picsum.photos/200/302"
  />
  <div>
    Haha
  </div>
</div>
`;
